### DISCUSSION TITLE
Issues when building MOOSE for the first time

### DISCUSSION THREAD

**Discussion by JoshWho-cpu** [idaholab/moose#25481] [Q&A Getting Started] [answered]
I was following the instructions at https://mooseframework.inl.gov/getting_started/installation/conda.html
This is what happened in my terminal:

(moose) root@(hidden, not sure if private):~/projects/moose# cd ~/projects/moose/test
(moose) root@(hidden):~/projects/moose/test# make -j 6
Rebuilding symlinks in /root/projects/moose/framework/build/header_symlinks
Creating Unity Directory /root/projects/moose/framework/build/unity_src
Copying default MOOSE configuration to: /root/projects/moose/framework/include/base/MooseConfig.h...
Creating Unity /root/projects/moose/framework/build/unity_src/outputs_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/outputs_png_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/outputs_formatters_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/dirackernels_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/fviks_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/interfaces_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/actions_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/multiapps_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/splits_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/partitioner_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/problems_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/base_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/controls_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/executioners_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/fvkernels_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/mesh_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/timesteppers_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/transfers_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/distributions_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/postprocessors_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/systems_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/userobjects_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/restart_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/geomsearch_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/scalarkernels_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/vectorpostprocessors_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/auxkernels_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/parser_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/meshgenerators_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/dgkernels_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/limiters_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/relationshipmanagers_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/functions_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/interfacekernels_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/dampers_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/constraints_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/utils_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/libtorch_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/libtorch_controls_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/libtorch_postprocessors_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/libtorch_utils_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/libtorch_reporters_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/samplers_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/ics_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/loops_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/timeintegrators_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/linesearches_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/preconditioners_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/markers_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/positions_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/kernels_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/materials_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/indicators_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/executors_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/bcs_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/fvbcs_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/times_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/nodalkernels_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/reporters_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/functormaterials_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/predictors_Unity.C
Creating Unity /root/projects/moose/framework/build/unity_src/variables_Unity.C
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/pcre/src/pcrecpp.cc...
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/pcre/src/pcre_stringpiece.cc...
Compiling C (in opt mode) /root/projects/moose/framework/contrib/pcre/src/pcre_tables.c...
Compiling C (in opt mode) /root/projects/moose/framework/contrib/pcre/src/pcre_get.c...
Compiling C (in opt mode) /root/projects/moose/framework/contrib/pcre/src/pcre_valid_utf8.c...
Compiling C (in opt mode) /root/projects/moose/framework/contrib/pcre/src/pcre_fullinfo.c...
Compiling C (in opt mode) /root/projects/moose/framework/contrib/pcre/src/pcre_byte_order.c...
Compiling C (in opt mode) /root/projects/moose/framework/contrib/pcre/src/pcre_jit_compile.c...
Compiling C (in opt mode) /root/projects/moose/framework/contrib/pcre/src/pcre_globals.c...
Compiling C (in opt mode) /root/projects/moose/framework/contrib/pcre/src/pcre_study.c...
Compiling C (in opt mode) /root/projects/moose/framework/contrib/pcre/src/pcre_dfa_exec.c...
Compiling C (in opt mode) /root/projects/moose/framework/contrib/pcre/src/pcre_maketables.c...
Compiling C (in opt mode) /root/projects/moose/framework/contrib/pcre/src/pcre_compile.c...
Compiling C (in opt mode) /root/projects/moose/framework/contrib/pcre/src/pcre_newline.c...
Compiling C (in opt mode) /root/projects/moose/framework/contrib/pcre/src/pcre_ucd.c...
Compiling C (in opt mode) /root/projects/moose/framework/contrib/pcre/src/pcre_version.c...
Compiling C (in opt mode) /root/projects/moose/framework/contrib/pcre/src/pcre_chartables.c...
Compiling C (in opt mode) /root/projects/moose/framework/contrib/pcre/src/pcre_config.c...
Compiling C (in opt mode) /root/projects/moose/framework/contrib/pcre/src/pcre_xclass.c...
Compiling C (in opt mode) /root/projects/moose/framework/contrib/pcre/src/pcre_exec.c...
Compiling C (in opt mode) /root/projects/moose/framework/contrib/pcre/src/pcre_string_utils.c...
Compiling C (in opt mode) /root/projects/moose/framework/contrib/pcre/src/pcre_refcount.c...
Compiling C (in opt mode) /root/projects/moose/framework/contrib/pcre/src/pcre_ord2utf8.c...
Compiling C++ /root/projects/moose/framework/contrib/gtest/gtest-all.cc...
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/hit/parse.cc...
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/hit/lex.cc...
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/hit/braceexpr.cc...
Building and linking /root/projects/moose/framework/contrib/hit/hit.so...
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/exodiff/ED_SystemInterface.C...
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/exodiff/Tolerance.C...
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/exodiff/STRINGLIB_tokenize.C...
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/exodiff/side_set.C...
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/exodiff/exodiff.C...
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/exodiff/util.C...
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/exodiff/SL_tokenize.C...
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/exodiff/stringx.C...
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/exodiff/create_file.C...
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/exodiff/exo_entity.C...
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/exodiff/GetLongOpt.C...
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/exodiff/FileInfo.C...
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/exodiff/check.C...
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/exodiff/add_to_log.C...
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/exodiff/iqsort.C...
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/exodiff/exoII_read.C...
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/exodiff/node_set.C...
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/exodiff/exo_block.C...
Compiling C++ (in opt mode) /root/projects/moose/framework/contrib/exodiff/map.C...
Creating Unity Directory /root/projects/moose/test/build/unity_src
Rebuilding symlinks in /root/projects/moose/test/build/header_symlinks
Creating Unity /root/projects/moose/test/build/unity_src/dirackernels_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/fviks_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/actions_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/partitioner_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/problems_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/controls_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/executioners_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/fvkernels_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/timesteppers_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/postprocessors_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/userobjects_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/meshes_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/scalarkernels_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/vectorpostprocessors_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/auxkernels_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/meshgenerators_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/dgkernels_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/functions_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/interfacekernels_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/constraints_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/libtorch_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/libtorch_vectorpostprocessors_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/samplers_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/ics_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/markers_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/kernels_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/materials_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/indicators_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/executors_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/bcs_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/fvbcs_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/nodalkernels_Unity.C
Creating Unity /root/projects/moose/test/build/unity_src/reporters_Unity.C
Creating Resource file /root/projects/moose/test/moose_test.yaml
Checking if header needs updating: /root/projects/moose/framework/include/base/MooseRevision.h...
Creating Unity /root/projects/moose/framework/build/unity_src/src_Unity.C
Symlinking MOOSE configure /root/projects/moose/framework/build/header_symlinks/MooseConfig.h
Linking Library /root/projects/moose/framework/contrib/pcre/libpcre-opt.la...
Linking Library /root/projects/moose/framework/contrib/gtest/libgtest.la...
Linking Library /root/projects/moose/framework/contrib/hit/libhit-opt.la...
Creating Unity /root/projects/moose/test/build/unity_src/outputs_Unity.C
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/dirackernels_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/fviks_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/actions_Unity.C...
make[1]: Entering directory '/root/projects/moose/framework/contrib/hit'
Building hit for python with python3-config
mpicxx -std=c++17 -g -fvisibility-inlines-hidden  -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /root/mambaforge3/envs/moose/include -std=c++17 main.cc parse.cc lex.cc braceexpr.cc -o hit
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/partitioner_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/problems_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/controls_Unity.C...
make[1]: Leaving directory '/root/projects/moose/framework/contrib/hit'
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/executioners_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/fvkernels_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/timesteppers_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/postprocessors_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/userobjects_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/meshes_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/scalarkernels_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/vectorpostprocessors_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/auxkernels_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/meshgenerators_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/dgkernels_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/functions_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/interfacekernels_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/constraints_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/libtorch_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/libtorch_vectorpostprocessors_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/samplers_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/ics_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/markers_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/kernels_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/materials_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/indicators_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/executors_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/bcs_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/fvbcs_Unity.C...
Compiling C++ (in opt mode) /root/projects/moose/test/build/unity_src/nodalkernels_Unity.C...
{standard input}: Assembler messages:
{standard input}:301312: Warning: end of file not at end of a line; newline inserted
{standard input}:302265: Error: invalid operands (*UND* and .gcc_except_table._ZNK7libMesh10Parameters3getISt6vectorIS2_I20MaterialPropertyNameSaIS3_EESaIS5_EEEERKT_St17basic_string_viewIcSt11char_traitsIcEE sections) for `-'
{standard input}:302295: Error: invalid operands (.text._ZNK7libMesh10Parameters3getISt6vectorIS2_I20MaterialPropertyNameSaIS3_EESaIS5_EEEERKT_St17basic_string_viewIcSt11char_traitsIcEE and *UND* sections) for `-'
x86_64-conda-linux-gnu-c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make: *** [/root/projects/moose/framework/build.mk:150: /root/projects/moose/test/build/unity_src/userobjects_Unity.x86_64-conda-linux-gnu.opt.lo] Error 1
make: *** Waiting for unfinished jobs....

I am using WSL ubuntu. I have installed MOOSE using conda. Running this command took about 15 minutes. Does anyone know what this message means and what I could do to fix it?

**Comment by GiudGiud** [comment, marked as answer]
The error seems pretty cryptic for a new install. 

I wonder if you run out of memory running -j6

Can you try to finish the compilation with make -j1

**Reply by JoshWho-cpu** [reply]
Thank you for reply, very happy that worked! It took about 25 minutes to build.

I am trying to run it with ./run_tests -j 1 and it has been going for about 30 minutes. Still printing out OK or SKIP, and I assume it will eventually finish successfully.

I assume the long wait time is because test likely goes through many cases to ensure that MOOSE is working as expected, and that most code I will write will not take as long.

Do you know where I can find documentation on the differences between -j1, -j6, -j... etc? I'm new to MOOSE and WSL so I am not sure what this means or where it comes from.

Editing to say it just finished after 1628.5 seconds.

**Reply by GiudGiud** [reply]
Hello

Yeah the install takes forever on WSL, and so does running the tests.
You would not have run out of memory running the tests. So you could have `./run_tests -j8` for example
You just dont get as much performance on WSL. We recommend using either a computing cluster or a strong laptop running linux or macos natively

Guillaume
